### PR TITLE
add relhum

### PR DESCRIPTION
**Pull request by cosunae**
## Purpose

Add a relative humidity operator. Notice that there are several implementation in FE reference, depending on the available input fields and 3D vs 2D. Only the code path required for implementing the meteo_friuli_venezia (cosmo-2e) was implemented.

## Code changes:

- Add relhum operator
- Add a relhum operator test

## Additional notes to reviewer

This PR belongs to a series to implement product structure based on dask delay. The order of dependency of these PRs follow: 

| idpi  |  idpi_products |
|------ | ----------------|
|grib_reader (#72)   ||
|first_product (#73)  | <----   first_product (https://github.com/MeteoSwiss-APN/idpi-products/pull/6)|
|                                |         support_various_datasets (https://github.com/MeteoSwiss-APN/idpi-products/pull/7)|
|relhum         (#74)   | <----   add_cosmo2e_products (https://github.com/MeteoSwiss-APN/idpi-products/pull/8)|

## Checklist
Before submitting this PR, please make sure:

- [x] You have followed the coding standards guidelines established at [Code Review Checklist](https://meteoswiss.atlassian.net/wiki/spaces/UA/pages/20157433/Factory).
- [x] Docstrings and type hints are added to new and updated routines, as appropriate
- [x] All relevant documentation has been updated or added (e.g. README)
- [x] Unit tests are added or updated for non-operator code
- [x] New operators are properly tested

## Review
For the review process follow the guidelines at [Checklist](https://meteoswiss.atlassian.net/wiki/spaces/UA/pages/20156488/Code+Review)

### CODE REVIEW

File: src/idpi/operators/atmo.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    return pc_rdv * pv / np.maximum((p - pc_o_rdv * pv), 1.0)
+
+
+def pv_qp(qv, p):
+    """Partial pressure of water vapor (from perfect gas law and approximating q~w).
+
+    Parameters
+    ----------
+    qv : xr.DataArray
+        waver vapor mixing ratio
```
Comment (cfkanesan) [inline src/idpi/operators/atmo.py:56]: water

File: src/idpi/operators/relhum.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+# Third-party
+import xarray as xr
+
+# First-party
+from idpi.operators.atmo import pv_sw
+from idpi.operators.atmo import qv_pvp
+
+
+def relhum(
+    qv, t, p, clipping=True, phase: Literal["water", "ice", "water+ic"] = "water"
```
Comment (cfkanesan) [inline src/idpi/operators/relhum.py:14]: water+ice

File: src/idpi/operators/relhum.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+):
+    """Calculate relative humidity.
+
+    Parameters
+    ----------
+    qv : xr.DataArray
+        waver vapor mixing ratio
```
Comment (cfkanesan) [inline src/idpi/operators/relhum.py:21]: water

File: src/idpi/operators/relhum.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    t : xr.DataArray
+        temperature
+    p : xr.DataArray
+        pressure
+    clipping : bool
+        clips the relative humidity to [0,100] interval
```
Comment (cfkanesan) [inline src/idpi/operators/relhum.py:27]: should note that only the upper bound is controlled by this parameter

File: tests/test_idpi/test_relhum.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+
+def test_relhum(data_dir, fieldextra):
+    datafile = data_dir / "lfff00000000.ch"
+    cdatafile = data_dir / "lfff00000000c.ch"
+
+    reader = GribReader([cdatafile, datafile], ref_param="P")
+    ds = reader.load_cosmo_data(["P", "T", "QV"])
+
+    brn = relhum(ds["QV"], ds["T"], ds["P"], clipping=True, phase="water")
```
Comment (cfkanesan) [inline tests/test_idpi/test_relhum.py:16]: brn?

File: tests/test_idpi/fieldextra_templates/test_RELHUM.nl [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+  in_regrid_target="GRID"
+  tstart=0, tstop=0, tincr=1
+  out_file="{{ file.output }}"
+  out_type="NETCDF"
+  in_size_field=570
+/
+&Process in_field="HSURF", use_tag="GRID", tag="hsurf" /
+&Process in_field="HEIGHT", use_tag="hhl_c1e", tag="HHL", level_class="k_half", levmin=1, levmax=81 /
+&Process in_field="HEIGHT", use_tag="hhl_c1e", tag="HFL", level_class="k_full", levmin=1, levmax=80 /
+&Process in_field="SOILTYP" /
```
Comment (cfkanesan) [inline tests/test_idpi/fieldextra_templates/test_RELHUM.nl:62]: Are these fields needed?

File: src/idpi/operators/relhum.py (hunk 2 above)
Comment (cosunae) [reply src/idpi/operators/relhum.py]: done

File: src/idpi/operators/atmo.py (hunk 1 above)
Comment (cosunae) [reply src/idpi/operators/atmo.py]: done

File: src/idpi/operators/relhum.py (hunk 4 above)
Comment (cosunae) [reply src/idpi/operators/relhum.py]: done

File: tests/test_idpi/test_relhum.py (hunk 5 above)
Comment (cosunae) [reply tests/test_idpi/test_relhum.py]: fixed

File: tests/test_idpi/fieldextra_templates/test_RELHUM.nl (hunk 6 above)
Comment (cosunae) [reply tests/test_idpi/fieldextra_templates/test_RELHUM.nl]: no, removed

File: src/idpi/constants.py [hunk 7]
```diff
@@ -7,7 +7,13 @@
...
+pc_rdv = pc_r_d / pc_r_v
+pc_o_rdv = 1.0 - pc_rdv
 pc_cp_d = 1005.0
 pc_rvd = pc_r_v / pc_r_d
 pc_rdocp = pc_r_d / pc_cp_d
 pc_rvd_o = pc_rvd - 1.0
+pc_b1 = 611.21  # Pressure at triple point of water [Pa]
+pc_b2w = 17.502
+pc_b3 = 273.16  # Temperature at triple point of water [K]
+pc_b4w = 32.19  # [K]
```
Comment (victoria-cherkas) [inline src/idpi/constants.py:19]: Should we not have descriptions for new constants?

Comment (cosunae) [reply src/idpi/constants.py:22]: thanks, yes. They are constants coming from the literature, I am not sure it is easy to add a description, but I added a reference where they can be found. (The same reference that fieldextra uses)

Comment (victoria-cherkas) [reply src/idpi/constants.py:22]: Perfect, thanks

**Review by victoria-cherkas — APPROVED** [review]
Approved, minus pre-commit failing

**Pull request merged**